src/odemis/gui/model/tab_gui_data.py

```diff
@@ -386,6 +397,49 @@ def _on_project_path_change(self, _):
                     config.fn_count)
 
 
+class CryoFIBSEMGUIData(CryoGUIData):
+    """ Represent an interface used to control the FIBSEM.
+    It it used for METEOR systems.
+    """
+
+    def __init__(self, main):
+        super().__init__(main)
+
+        # Current tool selected (from the toolbar)
+        tools = {TOOL_NONE, TOOL_RULER, TOOL_FEATURE}
+        # Update the tool selection with the new tool list
+        self.tool.choices = tools
+        # VA for autofocus procedure mode
+        self.autofocus_active = BooleanVA(False)
+        # the streams to acquire among all streams in .streams
+        self.acquisitionStreams = model.ListVA()
+        # the static overview map streams, among all streams in .streams
+        self.overviewStreams = model.ListVA()
+        # for the filename
+        config = conf.get_acqui_conf()
+        self.filename = model.StringVA(create_filename(
+            config.pj_last_path, config.fn_ptn,
+            config.last_extension,
+            config.fn_count))
+        self.main.project_path.subscribe(self._on_project_path_change)
+
+        # milling patterns
+        self.patterns = model.ListVA()
+
+        self.view_posture = model.VigilantAttribute(SEM_IMAGING)
+        self.acqui_mode = AcquiMode.FIBSEM
+        self.is_sem_active_view: bool = False
+        self.is_fib_active_view: bool = False
```

Comment:
They seems to detect which view is active. You can already do that with `self.focussedView.value is self.panel.vp_secom_tl.view`. => drop.

Reply:
when this is accessed, you don't necessarily have direct access to the tab or panel, so it's clearer to have the property

Reply:
The idea of the classes in the gui.model is that represent the state of the GUI. Their attributes should be "observable" (ie, they are VigilantAttributes) and information should be defined only once. 

So I'd recommend to move these out of the model.CryoFIBSEMGUIData . One alternative to my previous suggestion is to use the stream_classes of the view (which are SEMStream and FIBStream on the respective views). Something like:
`self.focussedView.value.is_compatible(SEMStream)` or `self.focussedView.value.is_compatible(FIBStream)`

Reply:
ok removed